.github/workflows/pr_checklists.yml

```diff
@@ -0,0 +1,109 @@
+            if [[ "$FILE" == *.html ]]; then
+              if grep -qE '<a\s+[^>]*href=|<button' "$FILE"; then
+                INCLUDE_ANALYTICS=true
+              fi
+            fi
+
+            # Check JS files for event listeners and experiment views
+            if [[ "$FILE" == *.js ]]; then
+              grep -qE '\.addEventListener|onclick|onchange|onsubmit' "$FILE" && INCLUDE_ANALYTICS=true
+              grep -q 'experiment_view' "$FILE" && INCLUDE_EXPERIMENT=true
+            fi
+
+            # Check if any views.py files have values in the entrypoint_variation allow list
+            if [[ "$FILE" == *views.py ]]; then
+              if grep -q 'entrypoint_variation' "$FILE"; then
+                VALUE=$(grep 'entrypoint_variation' "$FILE" | grep -v '\[\s*\]' || true)
+                [[ -n "$VALUE" ]] && INCLUDE_EXPERIMENT=true
+              fi
+            fi
+
+            # Stop looping through files if everything is already included
+            if [[ "$INCLUDE_CSS" == "true" && "$INCLUDE_ANALYTICS" == "true" && 
+                  "$INCLUDE_L10N" == "true" && "$INCLUDE_EXPERIMENT" == "true" && 
+                  "$INCLUDE_HTML" == "true" && "$INCLUDE_JS" == "true" ]]; then
+              # All flags are set, no need to check more files
+              break
+            fi
+
+          done < changed_files.txt
+
+          echo "INCLUDE_CSS=$INCLUDE_CSS" >> $GITHUB_ENV
+          echo "INCLUDE_ANALYTICS=$INCLUDE_ANALYTICS" >> $GITHUB_ENV
+          echo "INCLUDE_L10N=$INCLUDE_L10N" >> $GITHUB_ENV
+          echo "INCLUDE_EXPERIMENT=$INCLUDE_EXPERIMENT" >> $GITHUB_ENV
+          echo "INCLUDE_HTML=$INCLUDE_HTML" >> $GITHUB_ENV
+          echo "INCLUDE_JS=$INCLUDE_JS" >> $GITHUB_ENV
+
+      - name: Build comment body
+        id: comment
+        run: |
+          BODY="Thank you for your pull request. Based on the contents you may need to double-check these things:\n"
+
+          append_checklist() {
+            local title="$1"
+            local file="$2"
+            echo -e "\n---\n\n<details>\n<summary><strong>$title</strong></summary>\n\n$(cat $file)\n</details>\n" >> comment.txt
+          }
+
+          echo -e "$BODY" > comment.txt
+
+          [[ "$INCLUDE_CSS" == "true" ]] && append_checklist "🧾 CSS Checklist" "docs/checklists/css.md"
+          [[ "$INCLUDE_ANALYTICS" == "true" ]] && append_checklist "📊 Analytics Checklist" "docs/checklists/analytics.md"
+          [[ "$INCLUDE_L10N" == "true" ]] && append_checklist "🌍 L10n Checklist" "docs/checklists/l10n.md"
+          [[ "$INCLUDE_EXPERIMENT" == "true" ]] && append_checklist "🧪 Experiment Checklist" "docs/checklists/experiment.md"
+          [[ "$INCLUDE_HTML" == "true" ]] && append_checklist "🧱 HTML Checklist" "docs/checklists/html.md"
+          [[ "$INCLUDE_JS" == "true" ]] && append_checklist "📜 JavaScript Checklist" "docs/checklists/js.md"
+
+      - name: Post PR comment
+        uses: marocchino/sticky-pull-request-comment@v2
```

Comment:
We can only use [approved github actions in our own workflows](https://github.com/MoCo-GHE-Admin/Approved-GHE-add-ons/blob/main/New_Org_Default_settings.md), and this is currently not one of the approved ones. 

We would have to get it approved or fork it and take responsibility for its behaviour and security (which we've done in the past I think)

Reply:
Updated to remove the action and move some of the functionality into the workflow \o/